src/Web/WeightControl.UnitTests/Products/CreateProductsServiceTests.cs

```diff
@@ -90,17 +79,58 @@ public async Task Create_ShouldReturnProductDto()
                 .GetMock<IValidator<ProductDto>>()
                 .Verify(x => x.Validate(It.IsAny<ProductDto>()), Times.Once);
 
+            mocker
+                .GetMock<IRepository<Product>>()
+                .Verify(x => x.FindAsync(It.IsAny<Expression<Func<Product, bool>>>()), Times.Once);
+
+            mocker
+                .GetMock<IRepository<Product>>()
+                .Verify(x => x.CreateAsync(It.IsAny<Product>()), Times.Once);
+
             mocker
                 .GetMock<IMapper>()
                 .Verify(x => x.Map<ProductDto>(It.IsAny<Product>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task Create_ShouldReturnBadRequestException_IfProductDtoIsNotValid()
+        {
+            // Arrange
+            var validator = new ProductDtoValidator();
+
+            var actualProductDto = new ProductDto();
+
+            mocker
+                .GetMock<IValidator<ProductDto>>()
+                .Setup(x => x.Validate(It.IsAny<ProductDto>()))
+                .Returns(validator.Validate(actualProductDto));
```

Comment:
It is not cool...use `new ValidationResult(new List<ValidationFailure> { new() })`  insetad